### DISCUSSION TITLE
creating-timer-dismissable-notifications-visual-studio-code-extension/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#610] [Comments]
# creating-timer-dismissable-notifications-visual-studio-code-extension/

Azure / Microsoft 365 / Visual Studio Code / Speaker / Blogger

https://www.eliostruyf.com/creating-timer-dismissable-notifications-visual-studio-code-extension/

**Comment by kevinunger** [comment]
Neat! Great idea!

I've modified the code a bit to fit my needs (I justed needed a simple notificaiton that hides itself after x seconds without a progress bar and without link).
Incase someone is looking for that, here is the code:

```javascript

import * as vscode from "vscode";

export async function showNotification({ message, timeout = 5000 }: { message: string; timeout?: number }) {
    await vscode.window.withProgress(
        {
            location: vscode.ProgressLocation.Notification,
            cancellable: false
        },
        async (progress) => {
            progress.report({ increment: 100, message: `${message}` });
            await new Promise((resolve) => setTimeout(resolve, timeout));
        }
    );
}
```
